File: app/src/mocked/java/com/makingiants/todayhistory/utils/Transformer.java
package com.makingiants.todayhistory.utils;

import rx.Observable;
import rx.schedulers.Schedulers;

/**
 * Transforms to avoid setting each Schedulers on each call with inmediate Scheduler. (for tests)
 * Use:
 *
 * observable.compose(applyIoSchedulers())
 */
public class Transformer {
  public static <T> Observable.Transformer<T, T> applyIoSchedulers() {
    return observable -> observable.subscribeOn(Schedulers.immediate())
        .observeOn(Schedulers.immediate());
  }
}

File: app/src/mocked/kotlin/com/makingiants/todayhistory/utils/Transformer.kt
package com.makingiants.todayhistory.utils

import rx.Observable
import rx.schedulers.Schedulers

/**
 * Transforms to avoid setting each Schedulers on each call with immediate Scheduler. (for tests)
 * Use:

 * observable.compose(applyIoSchedulers())
 */
object Transformer {
    fun <T> applyIoSchedulers(): Observable.Transformer<T, T> = Observable.Transformer<T, T> {
        it.subscribeOn(Schedulers.immediate()).observeOn(Schedulers.immediate())
    }
}